### PR TITLE
Let secret-op handle pkcs12 stores

### PR DESCRIPTION
**Pull request by dervoeti**
# Description

Fixes https://github.com/stackabletech/nifi-operator/issues/502.

Keytool is used to set a password for the PKCS12 stores, because NiFi complains when no password is specified:
```
WARN [main] o.a.nifi.security.util.SslContextFactory Some truststore properties are populated (/stackable/server_tls/truststore.p12, ********, PKCS12) but not valid
Failed to start web server: Invalid nifi.web.https configuration in nifi.properties
```
At least I could not make it work without a password for the store.

Local Kuttl test for LDAP with TLS worked fine.

Not sure about the naming of `STACKABLE_SERVER_TLS_DIR`, I just used the same name that was used in the change for trino-operator. Might not be appropiate for NiFi.

Also: I don't think we need the random destination alias (as implemented in trino-operator) in this case, but please re-check this.

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [x] Changes are OpenShift compatible
- [x] CRD changes approved
- [x] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [ ] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [x] Code contains useful comments
- [x] (Integration-)Test cases added
- [x] Documentation added or updated
- [x] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

**Comment by dervoeti** [comment]
When https://github.com/stackabletech/secret-operator/pull/314 is merged, we can probably get rid of the two keytool commands. I think the emptyDir volume is still needed, since an LDAP tls certificate might be added to the truststore (we should not write to the secret-op mount). But we could simply `cp` truststore.p12 to the emptyDir volume.

File: rust/operator-binary/src/controller.rs [hunk 1]
```diff
@@ -13,13 +13,15 @@ use crate::{config, OPERATOR_NAME};
...
 use snafu::{OptionExt, ResultExt, Snafu};
+use stackable_nifi_crd::STACKABLE_SERVER_TLS_DIR;
 use stackable_nifi_crd::{
     authentication::resolve_authentication_classes, Container, CurrentlySupportedListenerClasses,
     NifiCluster, NifiConfig, NifiConfigFragment, NifiRole, NifiStatus, APP_NAME, BALANCE_PORT,
     BALANCE_PORT_NAME, HTTPS_PORT, HTTPS_PORT_NAME, MAX_NIFI_LOG_FILES_SIZE,
     MAX_PREPARE_LOG_FILE_SIZE, METRICS_PORT, METRICS_PORT_NAME, PROTOCOL_PORT, PROTOCOL_PORT_NAME,
     STACKABLE_LOG_CONFIG_DIR, STACKABLE_LOG_DIR,
 };
+use stackable_operator::builder::SecretFormat;
```
Comment (maltesander) [inline rust/operator-binary/src/controller.rs:24]: Can we move this down as well in the existing import?

**Review by maltesander — COMMENTED** [review]
> When [stackabletech/secret-operator#314](https://github.com/stackabletech/secret-operator/pull/314) is merged, we can probably get rid of the two keytool commands. I think the emptyDir volume is still needed, since an LDAP tls certificate might be added to the truststore (we should not write to the secret-op mount). But we could simply `cp` truststore.p12 to the emptyDir volume.

Agreed, after operator-rs is released we can just copy. We need the password annotation though otherwise we cannot add the ldap cert.

File: rust/crd/src/lib.rs [hunk 2]
```diff
@@ -46,6 +46,7 @@ pub const METRICS_PORT: u16 = 8081;
 
 pub const STACKABLE_LOG_DIR: &str = "/stackable/log";
 pub const STACKABLE_LOG_CONFIG_DIR: &str = "/stackable/log_config";
+pub const STACKABLE_SERVER_TLS_DIR: &str = "/stackable/server_tls";
```
[the remark this answers is not in the document]
Comment (dervoeti) [reply rust/crd/src/lib.rs]: Fixed

**Comment by maltesander** [comment]
> @maltesander Thank you very much for the feedback, implemented all your suggestions. I also found a bug in the "create reporting task" job, which is a python script that needs the CA certificate to verify the server's identity. The certificate itself is not present anymore and has to be extracted from the PKCS12 store, which is now handled by keytool before the script runs.

I think we can just mount the cert here directly since its an independent job/pod?

**Review by maltesander — APPROVED** [review]
Thanks, LGTM!

**Pull request merged**